core/src/main/java/org/elasticsearch/rest/action/cat/AbstractCatAction.java

```diff
@@ -54,6 +54,8 @@ public AbstractCatAction(Settings settings) {
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         boolean helpWanted = request.paramAsBoolean("help", false);
         if (helpWanted) {
+            // consume all request parameters for 'help' to prevent an IllegalArgumentException
+            request.params().keySet().forEach(request::param);
```

Comment:
As @rjernst says, this incredibly broken behavior. It invalidates all the work we did to reject unrecognized URL parameters, now it accepts any parameters including garbage parameters.